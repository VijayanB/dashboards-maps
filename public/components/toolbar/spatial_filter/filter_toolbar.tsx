/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiButton, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
import { FilterByPolygon } from './filter_by_polygon';
import { FILTER_DRAW_MODE, DrawFilterProperties } from '../../../../common';

interface SpatialFilterToolBarProps {
  setFilterProperties: (properties: DrawFilterProperties) => void;
  isDrawActive: boolean;
}

export const SpatialFilterToolbar = ({
  setFilterProperties,
  isDrawActive,
}: SpatialFilterToolBarProps) => {
  const onCancel = () => {
    setFilterProperties({
      mode: FILTER_DRAW_MODE.NONE,
    });
  };
  const filterIconGroups = (
    <EuiFlexItem>
      <FilterByPolygon setDrawFilterProperties={setFilterProperties} isDrawActive={isDrawActive} />
    </EuiFlexItem>
  );
  if (isDrawActive) {
    return (
      <EuiFlexGroup gutterSize="s">
        <EuiFlexItem grow={false}>
          <EuiButton fill size="s" onClick={onCancel}>
            {'Cancel'}
          </EuiButton>
        </EuiFlexItem>
        {filterIconGroups}
      </EuiFlexGroup>
    );
  }
  return (
    <EuiFlexGroup responsive={false} direction="column" alignItems="flexStart" gutterSize="s">
      {filterIconGroups}
    </EuiFlexGroup>
  );
};
