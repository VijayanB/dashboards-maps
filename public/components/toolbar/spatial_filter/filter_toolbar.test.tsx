/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { shallow } from 'enzyme';
import React from 'react';
import { SpatialFilterToolbar } from './filter_toolbar';

it('renders spatial filter before drawing', () => {
  const mockCallback = jest.fn();
  const toolbarComponent = shallow(
    <SpatialFilterToolbar setFilterProperties={mockCallback} isDrawActive={false} />
  );
  expect(toolbarComponent).toMatchSnapshot();
});

it('renders spatial filter while drawing', () => {
  const mockCallback = jest.fn();
  const toolbarComponent = shallow(
    <SpatialFilterToolbar setFilterProperties={mockCallback} isDrawActive={true} />
  );
  expect(toolbarComponent).toMatchSnapshot();
});
