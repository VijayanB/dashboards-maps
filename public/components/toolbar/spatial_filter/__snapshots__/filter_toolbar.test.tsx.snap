// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders spatial filter before drawing 1`] = `
<EuiFlexGroup
  alignItems="flexStart"
  direction="column"
  gutterSize="s"
  responsive={false}
>
  <EuiFlexItem>
    <FilterByPolygon
      isDrawActive={false}
      setDrawFilterProperties={[MockFunction]}
    />
  </EuiFlexItem>
</EuiFlexGroup>
`;

exports[`renders spatial filter while drawing 1`] = `
<EuiFlexGroup
  gutterSize="s"
>
  <EuiFlexItem
    grow={false}
  >
    <EuiButton
      fill={true}
      onClick={[Function]}
      size="s"
    >
      Cancel
    </EuiButton>
  </EuiFlexItem>
  <EuiFlexItem>
    <FilterByPolygon
      isDrawActive={true}
      setDrawFilterProperties={[MockFunction]}
    />
  </EuiFlexItem>
</EuiFlexGroup>
`;
